src/actions/channels.ts

```diff
@@ -1394,11 +1418,21 @@ export function favoriteChannel(channelId: string): ActionFunc {
 
         Client4.trackEvent('action', 'action_channels_favorite');
 
-        return savePreferences(currentUserId, [preference])(dispatch);
+        // Add the channel to the Favorites category, unless this has been called from the categories update logic
```

Comment:
why the different treatment? wouldn't this be a moveChannelToCategory from a default one?

Reply:
Ideally it would be, but I never got around to changing all the favouriting logic to use the new method.

I did have a ticket for that though: https://mattermost.atlassian.net/browse/MM-23989